push/views.py

```diff
@@ -1,12 +1,25 @@
-from rest_framework.decorators import api_view, permission_classes
+from rest_framework.decorators import api_view, permission_classes, parser_classes
 from rest_framework.permissions import AllowAny
 from django.http import JsonResponse
-# Create your views here.
+from rest_framework.parsers import JSONParser
+from .serializers import SubscriptionsSerializer
+
+
+def rearrange_subscription_data(subscription_data) -> dict:
+    rearranged_subscription_data = dict()
+    rearranged_subscription_data["endpoint"] = subscription_data["endpoint"]
+    rearranged_subscription_data["public_key"] = subscription_data["keys"]["p256dh"]
+    rearranged_subscription_data["auth_key"] = subscription_data["keys"]["auth"]
+    return rearranged_subscription_data
 
 
 @api_view(['POST'])
+@parser_classes([JSONParser])
 @permission_classes([AllowAny])
 def subscribe_client(request, *args, **kwargs):
-    # api logic to save the data and make client a subscriber goes here
-    message = "Test"
-    return JsonResponse({"response": message})
+    subscription_data = rearrange_subscription_data(request.data)
+    subscription_object = SubscriptionsSerializer(data=subscription_data)
```

Comment:
The serializer should follow what your request contract is. It should be first thing that you do in your controller.

Reply:
subscription_object is not an object. Its a serializer. Name it that way